tests/js/user-tests.js

```diff
@@ -398,6 +398,28 @@ module.exports = {
     TestCase.assertNull(app.currentUser);
   },
 
+  async testLinkCredentials() {
+    let app = new Realm.App(appConfig);
+    await logOutExistingUsers(app);
+
+    let user = await app.logIn(Realm.Credentials.anonymous());
+    let all = app.allUsers;
+    assertIsSameUser(all[user.id], user);
+    TestCase.assertArrayLength(user.identities, 1);
+    TestCase.assertEqual(user.identities[0].providerType, "anon-user");
+
+    const validEmail = randomVerifiableEmail();
+    const validPassword = "test1234567890";
+    await app.emailPasswordAuth.registerUser(validEmail, validPassword);
+
+    let credentials = Realm.Credentials.emailPassword(validEmail, validPassword);
+    await user.linkCredentials(credentials);
+    all = app.allUsers;
+    assertIsSameUser(all[user.id], user);
+    TestCase.assertArrayLength(user.identities, 2);
+    TestCase.assertArraysEqual(user.identities.map(identity => identity.providerType).sort(), ["anon-user", "local-userpass"]);
+  },
```

Comment:
Perhaps expand this test with two users:
```suggestion
  async testLinkCredentials() {
    const app = new Realm.App(appConfig);
    await logOutExistingUsers(app);

    const user1 = await app.logIn(Realm.Credentials.anonymous());
    assertIsSameUser(app.allUsers[user1.id], user1);
    TestCase.assertArrayLength(user1.identities, 1);
    TestCase.assertEqual(user1.identities[0].providerType, "anon-user");

    const validEmail1 = randomVerifiableEmail();
    const validPassword1 = "User1-1234567890";
    await app.emailPasswordAuth.registerUser(validEmail1, validPassword1);

    let credentials1 = Realm.Credentials.emailPassword(validEmail1, validPassword1);
    await user1.linkCredentials(credentials1);
    assertIsSameUser(app.allUsers[user1.id], user1);
    TestCase.assertArrayLength(user1.identities, 2);
    TestCase.assertArraysEqual(user1.identities.map(identity => identity.providerType).sort(), ["anon-user", "local-userpass"]);

    // TODO: remove this if reusing anonymous users are removed from OS
    await user1.logOut(); // important as we otherwise get user1 when logging in again, and will get the following error when trying to link:
    // "linking a local-userpass identity is not allowed when one is already linked"

    const user2 = await app.logIn(Realm.Credentials.anonymous());
    assertIsSameUser(app.allUsers[user2.id], user2);
    TestCase.assertArrayLength(user2.identities, 1);
    TestCase.assertEqual(user2.identities[0].providerType, "anon-user");

    const validEmail2 = randomVerifiableEmail();
    const validPassword2 = "User2-1234567890";
    await app.emailPasswordAuth.registerUser(validEmail2, validPassword2);

    let credentials2 = Realm.Credentials.emailPassword(validEmail2, validPassword2);
    await user2.linkCredentials(credentials2);
    assertIsSameUser(app.allUsers[user2.id], user2);
    TestCase.assertArrayLength(user2.identities, 2);
    TestCase.assertArraysEqual(user2.identities.map(identity => identity.providerType).sort(), ["anon-user", "local-userpass"]);

    TestCase.assertNotEqual(user1.id, user2.id);

    await user2.logOut();
  },
```